crates/transaction-emitter/src/lib.rs

```diff
@@ -679,50 +680,46 @@ impl<'t, 'd> TxnEmitter<'t, 'd> {
 }
 
 async fn retrieve_account_balance(
-    client: &JsonRpcClient,
+    client: &RestClient,
     address: AccountAddress,
 ) -> Result<Vec<AmountView>> {
-    let resp = client
-        .get_account(address)
-        .await
-        .map_err(|e| format_err!("[{:?}] get_accounts failed: {:?} ", client, e))?
-        .into_inner();
-    Ok(resp
-        .ok_or_else(|| format_err!("account does not exist"))?
-        .balances)
+    Ok(client
+        .get_account_balances(&address)
+        .await?
+        .into_inner()
+        .into_iter()
+        .map(|b| AmountView {
+            amount: b.amount,
+            currency: b.currency_code(),
+        })
+        .collect())
 }
 
 pub async fn execute_and_wait_transactions(
-    client: &JsonRpcClient,
+    client: &RestClient,
     account: &mut LocalAccount,
-    txn: Vec<SignedTransaction>,
+    txns: Vec<SignedTransaction>,
 ) -> Result<()> {
     debug!(
         "[{:?}] Submitting transactions {} - {} for {}",
         client,
-        account.sequence_number() - txn.len() as u64,
+        account.sequence_number() - txns.len() as u64,
         account.sequence_number(),
         account.address()
     );
 
-    // Batch submit all the txns
-    for txn_batch in txn.chunks(20) {
+    let pending_txns: Vec<Response<PendingTransaction>> =
+        try_join_all(txns.iter().map(|t| client.submit(t)))
+            .await
+            .context("submit transactions failed")?;
+
+    for pt in pending_txns {
         client
-            .batch(
-                txn_batch
-                    .iter()
-                    .map(MethodRequest::submit)
-                    .collect::<Result<_, _>>()?,
-            )
-            .await?
-            .into_iter()
-            .map(|r| r.and_then(|response| response.into_inner().try_into_submit()))
-            .collect::<Result<Vec<()>, _>>()
-            .context("failed to submit transactions")?;
+            .wait_for_transaction(&pt.into_inner())
+            .await
+            .context("wait for transactions failed")?;
```

Comment:
Just a note that a better conversion would be to handle all of these concurrently instead of sequentially using something like `join_all`. Depending on perf we may not need to do that now but eventually we should.

Reply:
This would also apply to anywhere we used to use batching.

Reply:
Wait for txn is different, we just need to wait for the txn took longest time to execute. 
So if first txn is executed we will be blocked at first txn wait, then the rest wait for txn is just one simple call and no wait, which I feel is better than fire concurrent calls for all txns and wait.